### PR TITLE
docs(chart): update demo page to chartjs v4

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description

Update demo page to compatible with chartjs v4.

Found that chartjs v4 requires [date adapter](https://github.com/chartjs/awesome#adapters) to support scale type time axes. Comparing on v2, it included with moment. So I pick [chartjs-adapter-date-fns](https://github.com/chartjs/chartjs-adapter-date-fns) dependency because we already have had date-fns.

Fixes # ([STG-200)](https://jira.refinitiv.com/browse/STG-200))

## Type of change

- [x] Refactor (improves code without changing its functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

### CODE REVIEW

File: packages/elements/src/chart/index.ts [hunk 1]
```diff
@@ -36,6 +36,8 @@ import type { ChartConfiguration, ChartOptions, UpdateMode, LegendItem } from 'c
 import type { Header } from '../header';
 import '../header/index.js';
 
+import 'chartjs-adapter-date-fns';
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/index.ts:41]: We might need to work on locale support for `ef-chart` as documented in "Configuration" section of https://www.npmjs.com/package/chartjs-adapter-date-fns?activeTab=readme

File: packages/elements/src/chart/__demo__/index.html [hunk 2]
```diff
@@ -1116,25 +1103,24 @@
         }
       };
 
-      /* eslint-disable no-unused-vars */
-      let addData = () => {
+      const addDataButton = document.getElementById('addNewDataSet');
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/__demo__/index.html]:
```suggestion
      const addDataButton = document.getElementById('addData');
```

**Review by wattachai-lseg — CHANGES_REQUESTED** [review]

File: packages/elements/src/chart/__demo__/index.html (hunk 2 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/__demo__/index.html]: Updated.

File: packages/elements/src/chart/index.ts (hunk 1 above)
Comment (wattachai-lseg) [reply packages/elements/src/chart/index.ts:41]: as EFv 6 doesn't provide builtin locale support, there is no plan to introduce this feature in this v7 release. Marking as resovled.

**Review by wattachai-lseg — APPROVED** [review]
LGTM

**Pull request merged**